p2panda-rs/src/operation/operation_fields.rs

```diff
@@ -47,6 +47,10 @@ pub enum OperationValue {
     /// Reference to a list of document views.
     #[serde(rename = "pinned_relation_list")]
     PinnedRelationList(PinnedRelationList),
+
+    /// Reference to a document's owner key group.
+    #[serde(rename = "owner")]
+    Owner(Relation),
```

Comment:
This is the main point of contention for me, I believe we want to use a `PinnedRelation` here. There are pros and cons to both, as I see it:

### owner is a `Relation`
#### pros
- any updates to the owner `KeyGroup` are automatically applied to this document so no extra actions are needed to update a documents owner once it's set.
#### cons
- updates to a documents' owner `KeyGroup` apply to the entire history of a document, this means we need to then introduce a mechanism for retaining earlier operations which might be incorrectly ignored during materialisation (this is the `limit` logic we understand well though).
- as a result of the mutable nature of the owner -> `KeyGroup` relationship as defined above, document views are no longer immutable.
  -  the same `DocumentViewId` could produce a different view on different nodes if they haven't got the same `KeyGroup` or `KeyGroupLimit` updates
  - a `DocumentViewId` could resolve to different values on the same node if the `KeyGroupLimit` changes

### owner is a `PinnedRelation`
#### pros
- once set an owner relation for a specific operation will never change, this means document views retain their "immutable" property.
#### cons
- updates to a `KeyGroup` now don't bubble up to all it'd "owned" documents.... this means we need a mechanism for noticing `KeyGroup` updates and including any changes in operations clients publish.

In terms of this PR I am happy for it to be merged as it is though, it doesn't include the `key_group_limit` logic yet, and we can explore how it is to use a `Relation` for the owner field in any case :+1: 

I have a feeling that any difference of opinion on this might boil down to whether we imagine `KeyGroups` as a moderation or authorisation layer. I see them as an authorisation layer, which to me demands the immutable properties you get from `PinnedRelation`. I can see if they were more about moderation, then the mutable nature of `Relation` and introducing `KeyGroupLimit` is more what would be needed. For me the moderation layer happens elsewhere (which we haven't defined).

Reply:
> does that mean that members who joined the group after a document's owner was set to that group are not authorised to make changes to that document?

Yes, but this isn't the primary goal, it's a side-effect of having immutable permissions, the new flow would be the same as before but with one change:

1. Group G1 has members [a,]
2. Group G1 creates a blog post document P1 with itself as the owner (PinnedRelation).
3. Public key b joins the group.
4. Public key b makes a change to P1 _which includes updating the `owner` field to point at the newest document view of  Group G1 (when they are a member)_

Reply:
I think I've come around on this a bit. I have come to like the idea of immutable views a lot. I would still like to avoid having changes on _all_ owned documents whenever the members of a key group change. 

Could we resolve owner fields with a mechanism similar to lenses? We would first build an immutable view from a view id. Then whenever the group's members change we store an updated materialised lens on that view, which filters operations according to group membership.